### PR TITLE
feat: create privacy settings UI

### PR DESCRIPTION
**Pull request by simonheys**
Adds privacy settings screen, underlying shared settings storage and hooks and makes services conditional on these settings.

https://user-images.githubusercontent.com/175607/175032684-c48f7883-0a5f-4c3e-a065-421472a7e7db.mov

### CODE REVIEW

File: packages/extension/src/ui/services/useStorageValue.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import { useCallback, useEffect, useMemo, useState } from "react"
+import browser from "webextension-polyfill"
+
+import { Storage } from "../../background/storage"
```
Comment (janek26) [inline packages/extension/src/ui/services/useStorageValue.ts:4]: ui should never implement from background
even if it's just a type

please move type to shared in this case

File: packages/extension/src/ui/services/useStorageValue.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    const localStorageChange = async (
+      changes: Record<string, browser.Storage.StorageChange>,
+    ) => {
+      const changedKeys = Object.keys(changes)
+      const valueDidChange = changedKeys.includes(storageKey)
+      if (valueDidChange) {
+        updateStoredValue()
+      }
+    }
+    browser.storage.onChanged.addListener(localStorageChange)
```
Comment (janek26) [inline packages/extension/src/ui/services/useStorageValue.ts:52]: this also mixes the concept of background and ui
please use our messaging system

File: packages/extension/src/ui/features/accountTokens/tokenPriceHooks.ts [hunk 3]
```diff
@@ -13,30 +13,40 @@ import {
...
     {
       refreshInterval: 5 * 60 * 1000 /** 5 minutes */,
     },
   )
-  return {
-    pricesData,
-    tokenData,
-  }
+  /** don't return any data unless Argent services are enabled */
+  const { value: privacyUseArgentServicesValue } = useSettingsValue(
```
[the remark this answers is not in the document]
Comment (simonheys) [reply packages/extension/src/ui/features/accountTokens/tokenPriceHooks.ts:38]: Agree this looks a little ugly, but this way it is triggered immediately when the user setting hook changes. `argentServicesFetcher` is just an async fetcher function (not hook) and won't respond to the setting changing. Possibly a separate exercise to revisit this later?

File: packages/extension/src/ui/services/useStorageValue.ts (hunk 1 above)
Comment (simonheys) [reply packages/extension/src/ui/services/useStorageValue.ts]: Refactored to use messaging

File: packages/extension/src/ui/features/accountTokens/tokenPriceHooks.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (simonheys) [reply packages/extension/src/ui/features/accountTokens/tokenPriceHooks.ts]: Reworked to a new `useConditionallyEnabledSWR` and tests for this - naming is hard

File: packages/extension/src/ui/services/useStorageValue.ts (hunk 2 above)
Comment (simonheys) [reply packages/extension/src/ui/services/useStorageValue.ts]: Refactored to use messaging

File: packages/extension/package.json [hunk 4]
```diff
@@ -4,6 +4,7 @@
   "main": "index.js",
   "license": "MIT",
   "devDependencies": {
+    "@serh11p/jest-webextension-mock": "^4.0.0",
```
[the remark this answers is not in the document]
Comment (simonheys) [reply packages/extension/package.json]: Removed

File: packages/extension/src/ui/services/swr.ts [hunk 5]
```diff
@@ -27,3 +28,32 @@ export const swrCacheProvider: Cache = {
...
+
+const fetcherDisabled: BareFetcher<any> = () => undefined
+
+/**
+ * `useSWR` with additional flag that will enable or disable calling the fetcher and clears data when disabled
+ *
+ * This is useful when data fetching may be dependent on a user setting
+ */
+
+export function useConditionallyEnabledSWR<Data = any, Error = any>(
```
Comment (janek26) [inline packages/extension/src/ui/services/swr.ts:42]: if the caching key is undefined SWR will skip execution
https://swr.vercel.app/docs/conditional-fetching#conditional

**Review by janek26 — APPROVED** [review]

**Pull request merged**